sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -232,16 +232,29 @@ Command *Scheduler::GraphBuilder::insertMemoryMove(MemObjRecord *Record,
 
   AllocaCommandBase *AllocaCmdSrc =
       findAllocaForReq(Record, Req, Record->MCurContext);
+  if (IsSuitableSubReq(Req) && !AllocaCmdSrc) {
```

Comment:
```suggestion
  if (!AllocaCmdSrc && IsSuitableSubReq(Req)) {
```
Checking pointer should be more cheap operation and should fail in most of scenarios.

Reply:
Applied